File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.rs
impl Solution {
pub fn length_of_longest_substring(s: String) -> i32 {
    let mut res = 0;

    let mut mp = HashMap::new();
    let mut l = 0;

    for (r, ch) in s.chars().enumerate() {
        if let Some(&prev) = mp.get(&ch)
            && prev >= l
        {
            l = prev + 1;
        }
        mp.insert(ch, r);
        res = res.max(r - l + 1);
    }

    res as i32
}
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
 public:
  int maxProfit(vector<int>& p) {
    int res = 0, l = 0, r = 0;
    while (r < p.size()) {
      res = max(res, p[r] - p[l]);
      if (p[l] > p[r]) {
        l++;
      } else {
        r++;
      }
    }
    return res;
  }
};
